[CORL-2545] Show singular site options for banning on single site organizations

nick-funk opened the pull request:
## What does this PR do?

Update ban modal dialog options for single site organizations so that it shows a single ban/unban option when there is only one site in the organization.

## What changes to the GraphQL/Database Schema does this PR introduce?

None

## Does this PR introduce any new environment variables or feature flags? 

No

## If any indexes were added, were they added to `INDEXES.md`?

No new indexes.

## How do I test this PR?

- Setup Coral so there is only one site listed under `Admin > Configure > Organization > Sites`
- Create a mod user and log in as them
- Go to `Community` and attempt to ban a user
- See that the flow is simply to ban the user outright, no `All sites` or `Specific sites` options
- Ban them
- Go to `Manage ban` on same user
- See that the flow is simply to unban the user outright
- Go to `Admin > Configure > Organization > Sites` and add a new site
- Go back to `Community` and attempt to ban user again
- See that the flow now has `All sites` and `Specific sites` options, with save button for ban
- Ban them
- Go to `Manage ban` on the same user
- See that the flow now has `All sites` and `Specific sites` options, with save button for unban
- Unban them
 
## How do we deploy this PR?

No special considerations for deploy.

nick-funk committed:
Update ban modal dialog options for single site organizations

kabeaty commented on src/core/client/admin/components/ModerateCard/ModerateCardContainer.tsx:
```diff
@@ -410,6 +410,7 @@ const ModerateCardContainer: FunctionComponent<Props> = ({
           moderationScopesEnabled={settings.multisite}
           userBanStatus={comment.author.status.ban}
           userRole={comment.author.role}
+          isMultisite={settings.multisite}
```
Since we are already passing through `settings.multisite` for `moderationScopesEnabled`, it would be nice to combine these and just pass whether it is multisite through once but use it for both purposes.

nick-funk replied:
Agreed, will fix

kabeaty commented on src/core/client/admin/components/UserStatus/BanModal.tsx:
```diff
@@ -237,7 +262,12 @@ const BanModal: FunctionComponent<Props> = ({
                     viewerIsOrgAdmin ||
                     (viewerIsScoped && !viewerIsSingleSiteMod)) && (
-                    <Flex className={styles.sitesToggle} spacing={5}>
+                    <Flex
```
What about just adding `isMultisite` to the conditional check that starts on line 262 for showing the All / Specific / No sites options? Then we wouldn't need to add these `hidden` styles.

nick-funk replied:
Fixed and incoming shortly

kabeaty commented on src/core/client/admin/components/UserStatus/BanModal.tsx:
```diff
@@ -307,15 +337,23 @@ const BanModal: FunctionComponent<Props> = ({
-                        Save
-                      </Button>
-                    </Localized>
+                    {isMultisite && (
```
This is somewhat of a personal preference, but I find it clearer to do 
```
isMultisite ? (<Button></Button>) : (<Button></Button>)
```

I also think it could make sense and be cleaner here to just create a separate component called something like `BanModalSubmitButton` here and then determine what text and localization id to use in that component based on whether it's multisite and what the user's ban status is, since those are the main differences. Then the `createSingleSiteSubmitButton` logic could move into that component too.

nick-funk replied:
Great suggestion, refactoring now

kabeaty commented on src/core/client/admin/components/UserStatus/BanModal.tsx:
```diff
@@ -55,8 +54,33 @@ interface Props {
+  } else {
+    return (
+      <Localized id="community-banModal-ban">
+        <Button type="submit" ref={lastFocusableRef} disabled={disabled}>
```
Looking at the issue, this button should be changed to use the `alert` color option and be red in this case. Can you please confirm and update?

nick-funk replied:
You're correct, I will fix this up!

kabeaty reviewed (changes requested):
Works as expected and looks good! I left a few comments throughout.

nick-funk committed:
Simplify UI logic for single/multi-site ban modals, fix alert styles

- Refactor the ban button styles/text into separate component
- Simplify the multisite/mod scopes boolean logic
- Set alert color styles for the single site ban/unban button

kabeaty commented on src/core/client/admin/components/UserStatus/BanModal.tsx:
```diff
@@ -52,20 +50,75 @@ interface Props {
+          type="submit"
+          ref={lastFocusableRef}
+          disabled={disabled}
+          color="alert"
```
Looking at the issue, I think the unban is still supposed to be green and not use the alert color.

nick-funk replied:
Fixed

kabeaty commented on src/core/client/admin/components/UserStatus/UserStatusChangeContainer.tsx:
```diff
@@ -289,7 +289,7 @@ const UserStatusChangeContainer: FunctionComponent<Props> = ({
           onClose={handleBanModalClose}
           onConfirm={handleUpdateBan}
-          moderationScopesEnabled={moderationScopesEnabled}
+          isMultisite={settings.multisite && moderationScopesEnabled}
```
I think that `moderationScopesEnabled` is just equal to `settings.multisite`, so I don't think we need to check both here.

nick-funk replied:
Fixed

kabeaty reviewed (changes requested):
Looking good! Thanks for the updates. Just left two small notes.

nick-funk committed:
Remove duplicate multisite prop for BanModal

nick-funk committed:
Make ban modal unban button green again

kodiakhq[bot] committed:
Merge branch 'develop' into fix/CORL-2545-single-site-ban-flow
